app/actions/pendingTransactionActions.js

```diff
@@ -0,0 +1,150 @@
+      transaction.vout.pop()
+      // eslint-disable-next-line
+      for (const send of transaction.vout) {
+        parsedData.push({
+          confirmations,
+          txid,
+          net_fee,
+          blocktime,
+          amount: send.value,
+          to: send.address,
+          // eslint-disable-next-line no-await-in-loop
+          asset: await findAndReturnTokenInfo(send.asset, net),
+        })
+      }
+    }
+  }
+  const sortedByBlockTime = (
+    a: ParsedPendingTransaction,
+    b: ParsedPendingTransaction,
+  ) => b.blocktime - a.blocktime
+  const sorted: Array<ParsedPendingTransaction> = parsedData.sort(
+    sortedByBlockTime,
+  )
+  return sorted
+}
+
+const getPendingTransactions = async (): Promise<PendingTransactions> =>
+  getStorage(STORAGE_KEY)
+
+const setPendingTransactions = async (
+  pendingTransactions: PendingTransactions,
+): Promise<any> => setStorage(STORAGE_KEY, pendingTransactions)
+
+export const pruneConfirmedOrStaleTransaction = async (
+  address: string,
+  txId: string,
+) => {
+  const storage = await getPendingTransactions()
+  if (Array.isArray(storage[address])) {
+    storage[address] = storage[address].filter(
+      transaction => transaction !== txId,
+    )
+  }
+  await setPendingTransactions(storage)
+}
+
+export const addPendingTransaction = createActions(
+  ID,
+  ({ address, txId }) => async (): Promise<void> => {
+    const transactions = await getPendingTransactions()
+
+    if (Array.isArray(transactions[address])) {
+      transactions[address].push(txId)
+    } else {
+      transactions[address] = []
+    }
+    await setPendingTransactions(transactions)
+  },
+)
+
+export const getPendingTransactionInfo = createActions(
+  ID,
+  ({ address, net }) => async (): Promise<Array<Object>> => {
+    const transactions = await getPendingTransactions()
+    if (Array.isArray(transactions[address]) && transactions[address].length) {
+      let url = await getNode(net)
+      if (isEmpty(url)) {
+        url = await getRPCEndpoint(net)
+      }
+      const client = Neon.create.rpcClient(url)
+
+      const pendingTransactionInfo = []
+
+      // eslint-disable-next-line
+      for (const transaction of transactions[address]) {
+        if (transaction) {
+          // eslint-disable-next-line
+          const result = await client
+            .getRawTransaction(transaction, 1)
+            .catch(async e => {
+              console.error(
+                e,
+                'An transaction was added to storage that the blockchain does not recognize - purging from storage',
+              )
+              await pruneConfirmedOrStaleTransaction(address, transaction)
+            })
+
+          if (result.confirmations > MINIMUM_CONFIRMATIONS) {
+            // eslint-disable-next-line
+            await pruneConfirmedOrStaleTransaction(address, transaction)
+          }
+          pendingTransactionInfo.push(result)
```

Comment:
Shouldn't already pruned (stale/confirmed) transactions be excluded from this array?

Reply:
Yes thanks!